spec/models/saved_form_spec.rb

```diff
@@ -2,10 +2,63 @@
 
 RSpec.describe SavedForm, type: :model do
   it { should validate_presence_of(:service_slug) }
-  it { should validate_presence_of(:user_id) }
 
   it 'should set the default values' do
     expect(subject.attempts).to eq(0)
     expect(subject.active).to eq(true)
   end
+
+  context '#invalidated' do
+    let(:model) { SavedForm.new(user_id: '123', user_token: '456', user_data_payload: 'q1 => answer', secret_answer: 'hello', active: true, service_slug: 'some_slug') }
+    
+    it 'should consider the model invalidated if user id is empty' do
```

Comment:
```suggestion
    context 'if user id is empty' do
      it 'should consider the model invalidated' do
        # assertion
      end
    end
```

This is mainly for readability and keeping it similar to the other spec